CompilerSource/compiler/compile.cpp

```diff
@@ -732,9 +732,9 @@ wto << "namespace enigma_user {\nstring shader_get_name(int i) {\n switch (i) {\
 
     // Restore the compilers original working directory.
     #if CURRENT_PLATFORM_ID == OS_WINDOWS
-    SetCurrentDirectory(prevdir);
+    SetCurrentDirectory(prevdir.data());
     #else
-    chdir(prevdir);
+    chdir(prevdir.data());
     #endif
```

Comment:
@JoshDreamland can you please explain to me, just for my own curiosity, what was wrong with this code? I tried digging to find why it was crashing but I don't see any. The only thing I can think of is that Travis CI deleted the current working directory when it was called or for some other reason it did not exist.

Reply:
I am not sure that this fixed the issue, but the reason I told fundies to try it was that 4KiB is a lot of data to just throw on the stack. Typically the stack is about a megabyte in size, but I figured Travis might be running us with a much smaller version to conserve resources.